modules/Auth/reservation/reservationImpl.cpp

```diff
@@ -12,24 +12,37 @@ void reservationImpl::init() {
 void reservationImpl::ready() {
 }
 
-types::reservation::ReservationResult
-reservationImpl::handle_reserve_now(int& connector_id, types::reservation::Reservation& reservation) {
+types::reservation::ReservationResult reservationImpl::handle_reserve_now(types::reservation::Reservation& request) {
     // your code for cmd reserve_now goes here
 
-    const auto reservation_result = this->mod->auth_handler->handle_reservation(connector_id, reservation);
+    EVLOG_debug << "Handle reservation for evse id " << (request.evse_id.has_value() ? request.evse_id.value() : -1);
+
+    const auto reservation_result = this->mod->auth_handler->handle_reservation(request);
     if (reservation_result == ReservationResult::Accepted) {
-        this->mod->auth_handler->call_reserved(connector_id, reservation.reservation_id);
+        if (!this->mod->auth_handler->call_reserved(request.reservation_id, request.evse_id)) {
+            return ReservationResult::Rejected;
```

Comment:
Well that was because you said the evse manager always has the last saying about the reservation. So if the reservation is made but the evse manager sais it is not possible, it is cancelled again.

I don't like that behaviour as well.

Reply:
Yes but I can not see where the reservation is canceled if the evse manager rejects it